airflow/providers/amazon/aws/links/emr.py

```diff
@@ -54,9 +54,15 @@ def get_log_uri(
             "Requires either the output of a describe_cluster call or both an EMR Client and a job_flow_id."
         )
 
+    log_uri = None
+    parsed_log_uri = None
     if cluster:
-        log_uri = S3Hook.parse_s3_url(cluster["Cluster"]["LogUri"])
+        parsed_log_uri = S3Hook.parse_s3_url(cluster["Cluster"]["LogUri"])
     else:
         response = emr_client.describe_cluster(ClusterId=job_flow_id)
-        log_uri = S3Hook.parse_s3_url(response["Cluster"]["LogUri"])
-    return "/".join(log_uri)
+        if "LogUri" in response["Cluster"]:
+            parsed_log_uri = S3Hook.parse_s3_url(response["Cluster"]["LogUri"])
+
+    if parsed_log_uri:
+        log_uri = "/".join(parsed_log_uri)
+    return log_uri
```

Comment:
When cluster is provided, we get it by the same method but we provide it to avoid querying the API twice, so we can have the same exception.
We can simplify this method by using this code:
```suggestion
    cluster_info = (cluster or emr_client.describe_cluster(ClusterId=job_flow_id))["Cluster"]
    if "LogUri" not in cluster_info:
        return
    log_uri = S3Hook.parse_s3_url(cluster_info["LogUri"])
    return "/".join(log_uri)
```

Reply:
Thanks for your suggestion! I just updated it